src/packages/plan/plan.ts

```diff
@@ -0,0 +1,352 @@
+                }),
+                headers,
+                method: 'POST',
+              });
+
+              const createdActivityDirectiveData = (await createdActivityDirectiveResponse.json()) as {
+                data: {
+                  insert_activity_directive_one: ActivityDirective;
+                };
+              } | null;
+
+              if (createdActivityDirectiveData) {
+                const {
+                  data: { insert_activity_directive_one: createdActivityDirective },
+                } = createdActivityDirectiveData;
+                activityRemap[id] = createdActivityDirective.id;
+              }
+            },
+          ),
+        );
+
+        // remap all the anchor ids to the newly created activity directives
+        logger.info(`POST /importPlan: Re-assigning anchors: ${name}`);
+        await Promise.all(
+          activities.map(async ({ anchor_id: anchorId, id }) => {
+            if (anchorId !== null && activityRemap[anchorId] != null && activityRemap[id] != null) {
+              logger.info(
+                `POST /importPlan: Re-assigning anchor ${anchorId} to ${activityRemap[anchorId]} for activity ${activityRemap[id]}: ${name}`,
+              );
+              const activityDirectiveSetInput: ActivityDirectiveSetInput = {
+                anchor_id: activityRemap[anchorId],
+              };
+
+              return fetch(GQL_API_URL, {
+                body: JSON.stringify({
+                  query: gql.UPDATE_ACTIVITY_DIRECTIVE,
+                  variables: {
+                    activityDirectiveSetInput,
+                    id: activityRemap[id],
+                    plan_id: (createdPlan as PlanSchema).id,
+                  },
+                }),
+                headers,
+                method: 'POST',
+              });
+            }
+          }),
+        );
+
+        // associate the tags with the newly created plan
+        logger.info(`POST /importPlan: Importing plan tags: ${name}`);
+        const parsedTags: number[] = JSON.parse(tags);
+
+        const tagsInsert: PlanTagsInsertInput[] = parsedTags.map(tagId => ({
+          plan_id: (createdPlan as PlanSchema).id,
+          tag_id: tagId,
+        }));
+
+        await fetch(GQL_API_URL, {
+          body: JSON.stringify({ query: gql.CREATE_PLAN_TAGS, variables: { tags: tagsInsert } }),
+          headers,
+          method: 'POST',
+        });
+
+        logger.info(`POST /importPlan: Imported plan: ${name}`);
+      }
+      res.json(createdPlan);
+    } else {
+      throw Error('Plan creation unsuccessful.');
+    }
+  } catch (error) {
+    logger.error(`POST /importPlan: Error occurred during plan ${name} import`);
+    logger.error(error);
+
+    if (createdPlan) {
+      await fetch(GQL_API_URL, {
+        body: JSON.stringify({ query: gql.DELETE_PLAN, variables: { id: createdPlan.id } }),
+        headers,
+        method: 'POST',
+      });
+    }
+    res.send(null);
```

Comment:
Blocking: send an HTTPS error code in the event of a failure.

```suggestion
    res.send(500);
```